### PR TITLE
Clang format develop

### PR DESCRIPTION
**Pull request by brianlheim**
**Please read carefully.**

This is the PR to format develop. I have triple-checked everything here, including the .clang-format file itself, but please if you have time, check it again.

**This PR should be merged via rebase, not merge commit.**

The next steps should be:

(step zero, review this PR and 3.10 PR)

1. Merge this PR **via rebase**.
2. Merge the 3.10 PR **via rebase**.
3. Merge 3.10 into develop.
4. Create two new tags, `tag-clang-format-3.10` on the clang-format commit for 3.10, and `tag-clang-format-develop` on the clang-format commit for develop. Push them here. Delete any releases from CI.

Questions/comments welcome. I would like to merge this by the end of the day.

### CODE REVIEW

**Comment by lnihlen** [comment]
🥳🥳🥳

File: QtCollider/QObjectProxy.cpp [hunk 1]
```diff
@@ -37,548 +37,500 @@
...
+    connect(qObject, SIGNAL(destroyed(QObject*)), this, SLOT(invalidate()));
+    qObject->installEventFilter(this);
+    _eventHandlers.reserve(10);
 }
 
-QObjectProxy::~QObjectProxy()
-{
-  qcProxyDebugMsg( 1, QStringLiteral("Proxy is being deleted.") );
-}
+QObjectProxy::~QObjectProxy() { qcProxyDebugMsg(1, QStringLiteral("Proxy is being deleted.")); }
```
Comment (jrsurge) [inline QtCollider/QObjectProxy.cpp]: should this all be one line? looks a little odd

**Comment by lnihlen** [comment]
Scrolled through a good chunk of it, LGTM.

File: server/plugins/BeatTrack2.cpp [hunk 2]
```diff
@@ -18,569 +18,681 @@
...
+// 6 Nov 2007
 
 #include "ML.h"
 
-//need to add bestgroove option to store groove, else remove output which is currently always straight 16ths
+// need to add bestgroove option to store groove, else remove output which is currently always straight 16ths
+
+
+static const int g_numtempi = 120;
+static float g_periods[g_numtempi] = { 1,
```
Comment (jrsurge) [inline server/plugins/BeatTrack2.cpp]: this doesn't look ideal?

File: server/plugins/KeyTrack.cpp [hunk 3]
```diff
@@ -18,301 +18,1226 @@
...
-
-	//unit->m_frame=0;
-
-	OUT0(0) = 0.f;
-
-	unit->mCalcFunc = (UnitCalcFunc)&KeyTrack_next;
+// weighting parameters
+
+// 4096 FFT at 44100 SR
+static float g_weights44100[720] = {
```
Comment (jrsurge) [inline server/plugins/KeyTrack.cpp]: is this ok?

File: server/plugins/BeatTrack2.cpp (hunk 2 above)
Comment (jrsurge) [inline server/plugins/BeatTrack2.cpp]: is this ok?

File: server/plugins/BeatTrack.cpp [hunk 4]
```diff
@@ -44,836 +44,824 @@
...
+// this data assumes LAGS is 128
+static float g_m[128] = {
```
Comment (jrsurge) [inline server/plugins/BeatTrack.cpp]: is this ok?

File: server/plugins/MFCC.cpp [hunk 5]
```diff
@@ -18,245 +18,3479 @@
...
-	Clear(unit->m_numcoefficients, unit->m_mfcc);
-	for (int j=0; j<unit->m_numcoefficients; ++j)
-		ZOUT0(j) = 0.f;
-
-	unit->mCalcFunc = (UnitCalcFunc)&MFCC_next;
+// The plugin could be generalised to pass alternative scale data (corresponding to Barks, ERBs and different FFT sizes
+// etc) via a buffer For the moment, the data below is used and only 1024 FFT supported. For the associated Mel scale
+// spacing generation code see the bottom of the MFCC help file
+
+int g_startbin44100[42] = { 0,   2,   4,   5,   7,   9,   11,  13,  15,  17,  20,  23,  26,  29,
```
Comment (jrsurge) [inline server/plugins/MFCC.cpp]: this file might need checked

File: server/plugins/BeatTrack.cpp (hunk 4 above)
Comment (brianlheim) [reply server/plugins/BeatTrack.cpp]: i don't see anything wrong; if anything it's more readable than before with columns

**Comment by brianlheim** [comment]
The long lookup tables/number lists are the least of our concerns -- nobody is likely to ever maintain those aside from replacing or removing an entire table. whatever clang-format wants to do with them is fine with me

**Comment by jrsurge** [comment]
> The long lookup tables/number lists are the least of our concerns -- nobody is likely to ever maintain those aside from replacing or removing an entire table. whatever clang-format wants to do with them is fine with me

this is all fine for me, just wanted to gauge wider approval.
Happy to proceed with a merge

**Comment by brianlheim** [comment]
> this is all fine for me, just wanted to gauge wider approval.
> Happy to proceed with a merge

Thanks for pointing it out and for giving it such a thorough review!

**Pull request merged**